src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/json/JsonConvert.java

```diff
@@ -71,26 +77,72 @@ private static Path startApartments() {
    */
   @SuppressWarnings("serial")
   public static List<Apartment> jsonToApartments(Path jsonPath) throws IOException {
+    JsonbConfig config = new JsonbConfig().withAdapters(JsonConvert.getAdapter());
     String jsonString = Files.readString(jsonPath);
-    List<Apartment.Builder> apartmentsBuild;
     List<Apartment> apartments = new ArrayList<>();
     LOGGER.info("Create ArrayList of Apartment");
-
-    try (Jsonb jsonb = JsonbBuilder.create()) {
+    try (Jsonb jsonb = JsonbBuilder.create(config)) {
       LOGGER.info("Create Json builder");
-      apartmentsBuild =
+      apartments =
           jsonb.fromJson(
-              jsonString, new ArrayList<Apartment.Builder>() {}.getClass().getGenericSuperclass());
+              jsonString,
+              new ArrayList<Apartment>() {
+                /**/
+              }.getClass().getGenericSuperclass());
     } catch (Exception e) {
       throw new IllegalArgumentException(e);
     }
-
-    for (Builder apartmentToBuild : apartmentsBuild) {
-      apartments.add(apartmentToBuild.build());
-    }
     return apartments;
   }
 
+  /**
+   * A method that implement a {@link JsonbAdapter} to avoid the use of ApartmentBuilder Code
+   * inspired from the example in the course of Jean-Michel Doudoux available <a>
+   * href="https://www.jmdoudoux.fr/java/dej/chap-json-b.htm">here</a>
+   */
+  public static JsonbAdapter<Apartment, JsonObject> getAdapter() {
+    return new JsonbAdapter<>() {
+      @Override
+      public JsonObject adaptToJson(Apartment apart) {
+        return Json.createObjectBuilder()
```

Comment:
Il est très inélégant de répéter tous ces noms de champs ici, ce qui désactive essentiellement tout ce qu’apporte jsonb. Convertissez plutôt avec votre adapter depuis Builder vers Apartment, en laissant jsonb sérialiser dans un builder. (L’autre partie de l’adaptateur, dans l’autre sens, ne devrait pas vous être utile et peut donc renvoyer une UnsupportedOperationException.)

Reply:
Fix in aa1b18e94633f14bbf92bea8542ff07cc491c01e